presto-cli/src/test/java/io/prestosql/cli/TestInsecureQueryRunner.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.cli;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.json.JsonCodec;
+import io.airlift.units.Duration;
+import io.prestosql.client.ClientSession;
+import io.prestosql.client.ClientTypeSignature;
+import io.prestosql.client.Column;
+import io.prestosql.client.QueryResults;
+import io.prestosql.client.StatementStats;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+
+import java.io.FileInputStream;
+import java.io.PrintStream;
+import java.security.KeyStore;
+import java.security.SecureRandom;
+import java.time.ZoneId;
+import java.util.Locale;
+import java.util.Optional;
+
+import static com.google.common.io.ByteStreams.nullOutputStream;
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static com.google.common.net.HttpHeaders.LOCATION;
+import static com.google.common.net.HttpHeaders.SET_COOKIE;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static io.prestosql.cli.ClientOptions.OutputFormat.CSV;
+import static io.prestosql.client.ClientStandardTypes.BIGINT;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.testng.Assert.assertEquals;
+
+@Test(singleThreaded = true)
+public class TestInsecureQueryRunner
+{
+    private static final JsonCodec<QueryResults> QUERY_RESULTS_CODEC = jsonCodec(QueryResults.class);
+
+    private MockWebServer server;
+
+    @BeforeMethod
+    public void setup() throws Exception
+    {
+        server = new MockWebServer();
+        SSLContext sslContext = buildTestSslContext();
+        server.useHttps(sslContext.getSocketFactory(), false);
+        server.start();
+    }
+
+    private SSLContext buildTestSslContext() throws Exception
+    {
+        // Load self-signed certificate
+        String path = getClass().getClassLoader().getResource("insecure-ssl-test.jks").getPath();
```

Comment:
We shouldn't depend on the resource URL being a file path. Instead, ask for the input stream:
```java
KeyStore serverKeyStore = ...;
try (InputStream in = getClass().getResourceAsStream("/insecure-ssl-test.jks")) {
    serverKeyStore.load(in, ...);
}